src/Device/Driver/LX_EOS/LXEosDevice.cpp

```diff
@@ -0,0 +1,199 @@
+   * as LXWP2, so choosing odd numbers to reduce overlaps
+   *
+   * LXWP2 is also sent automatically whenever vario settings are changed by
+   * pilot. Periodical sending is there only as a backup sync method.
+   */
+
+  // Set intervals of periodic messages
+  PortWriteNMEA(port, "PFLX0,LXWP0,1,LXWP1,7,LXWP2,5,LXWP3,3", env);
+  port.Flush();
+
+  return true;
+}
+
+bool
+LXEosDevice::PutMacCready(double mc, OperationEnvironment& env)
+{
+  const std::lock_guard lock{ settings_mutex };
+  vario_settings.mc = mc;
+  return SendNewSettings(env);
+}
+
+bool
+LXEosDevice::PutBallast([[maybe_unused]] double fraction,
+                        double overload,
+                        OperationEnvironment& env)
+{
+  const std::lock_guard lock{ settings_mutex };
+  if (vario_settings.dry_mass_to_ref_mass_uptodate) {
+    vario_settings.bal = overload * vario_settings.dry_mass_to_ref_mass;
+    return SendNewSettings(env);
+  }
+  return false;
+}
+
+void
+LXEosDevice::OnCalculatedUpdate([[maybe_unused]] const MoreData& basic,
+                                const DerivedInfo& calculated)
+{
+  const std::lock_guard lock{ settings_mutex };
+  auto& old_polar = vario_settings.xcsoar_polar;
+  auto& new_polar = calculated.glide_polar_safety;
+  if (!ComparePolarCoefficients(old_polar.GetCoefficients(),
+                                new_polar.GetCoefficients()) ||
+      old_polar.GetDryMass() != new_polar.GetDryMass() ||
+      old_polar.GetReferenceMass() != new_polar.GetReferenceMass()) {
+    vario_settings.xcsoar_polar = calculated.glide_polar_safety;
+    CalculateDryMassToReferenceMassRatio(vario_settings);
+  }
+}
+
+bool
+LXEosDevice::PutBugs(double bugs, OperationEnvironment& env)
+{
+  const std::lock_guard lock{ settings_mutex };
+  vario_settings.bugs = (1.0 - bugs) * 100.0;
+  return SendNewSettings(env);
+}
+
+bool
+LXEosDevice::ParseNMEA(const char* String, NMEAInfo& info)
+{
+  if (!VerifyNMEAChecksum(String))
+    return false;
+
+  NMEAInputLine line(String);
+
+  const auto type = line.ReadView();
+  if (type == "$LXWP0"sv)
+    return LXWP0(line, info);
+  else if (type == "$LXWP1"sv) {
+    LXDevice::LXWP1(line, info.device);
+    altitude_offset.reliable = HasReliableAltOffset(info.device);
+    return true;
+  } else if (type == "$LXWP2"sv)
+    return LXWP2(line, info);
+  else if (type == "$LXWP3"sv)
+    return LXWP3(line, info);
+  else
+    return false;
+}
+
+bool
+LXEosDevice::SendNewSettings(OperationEnvironment& env)
+{
+  if (vario_settings.uptodate == true) {
+    char buffer[64];
+    snprintf(buffer,
+             64,
+             "PFLX2,%.1f,%.2f,%.0f,,,,,",
+             vario_settings.mc,
+             vario_settings.bal,
+             vario_settings.bugs);
+
+    PortWriteNMEA(port, buffer, env);
+    return true;
+  } else
+    return false;
+}
+
+bool
+LXEosDevice::HasReliableAltOffset(DeviceInfo device)
+{
+  if (device.product == "LX Eos"sv) {
+    if (device.software_version == "1.7"sv)
```

Comment:
Only Version 1.7 or equal or bigger then?

Reply:
I would even consider refusing service on a non supported firmware, instead of working around bugs on the device.